tools/sr-snapshots/Sources/Git/GitClient.swift

```diff
@@ -0,0 +1,113 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import Files
+import Shell
+import Foundation
+
+public struct GitClientError: Error, CustomStringConvertible {
+    public var description: String
+}
+
+public protocol GitClient {
+    func cloneIfNeeded(to directory: URL) throws
+    func pull() throws
+    func commit(message: String) throws
+    func push() throws
+}
+
+public struct NOPGitClient: GitClient {
+    public init() {}
+    public func cloneIfNeeded(to directory: URL) throws {}
+    public func pull() throws {}
+    public func commit(message: String) throws {}
+    public func push() throws {}
+}
+
+/// Naive implementation of basic git client.
+///
+/// It executes git commands directly in shell and requires `gh`CLI to be preinstalled on host (https://cli.github.com/)
+/// and authorised for cloning private repositories.
```

Comment:
The requirement of [`gh` CLI](https://cli.github.com/) might look quite hard, but that tool is already [pre-installed](https://github.com/DataDog/dd-sdk-ios/blob/cd8724314d49db8ed3f755560811bee2ea69ae83/Makefile#LL61C1-L61C47) and authorised for internal repo clones in our CI. It is also leveraged for few other automations, e.g. clone [datadog-ios](https://github.com/DataDog/dd-sdk-ios/blob/cd8724314d49db8ed3f755560811bee2ea69ae83/tools/distribution/src/dogfood/repository.py#L25-L38) for dogfooding.